pkg/proxy/proxy.go

```diff
@@ -305,11 +279,55 @@ func fillEgressDestinationInfo(info *accesslog.EndpointInfo, ipstr string) {
 	}
 }
 
+// fillIngressSourceInfo fills the EndpointInfo fields, by fetching
+// the consumable from the consumable cache of endpoint using identity sent by
+// source. This is needed in ingress proxy while logging the source endpoint
+// info.  Since there will be 2 proxies on the same host, if both egress and
+// ingress policies are set, the ingress policy cannot determine the source
+// endpoint info based on ip address, as the ip address would be that of the
+// egress proxy i.e host.
+func fillIngressSourceInfo(info *accesslog.EndpointInfo,
+	ip *net.IP, srcIdentity uint32) {
+
+	if srcIdentity != 0 {
+		if ip != nil {
+			if ip.To4() != nil {
+				info.IPv4 = ip.String()
+			} else {
+				info.IPv6 = ip.String()
+			}
+		}
+		fillIdentity(info, policy.NumericIdentity(srcIdentity))
+	} else {
+		// source security identity 0 is possible when somebody else other than
+		// the BPF datapath attempts to
+		// connect to the proxy.
+		// We should try to resolve if the identity belongs to reserved_host
+		// or reserved_world.
+		if ip != nil {
```

Comment:
`fillEndpointInfo()` already checks `ip` for `nil`. One of the checks can be removed.

Reply:
done. Removed additional check in `fillEndpointInfo()`